Set correct GNU ld emulation type for illumos x86_64 binaries. This prevents
/usr/gnu/bin/ld: relocatable linking with relocations from format elf64-x86-64-sol2 (eu_braille.o) to format elf32-i386-sol2 (braille.o) is not supported

--- brltty-6.0/config.mk.in.orig	2019-02-03 19:03:48.000000000 +0000
+++ brltty-6.0/config.mk.in	2021-08-17 17:22:32.475965295 +0000
@@ -263,7 +263,7 @@
 LDLIBS = $(ICU_LIBS) $(SYSTEM_LIBS) @LIBS@
 
 MKSHR = @MKSHR@
-MKREL = @MKREL@
+MKREL = /usr/gnu/bin/ld -melf_x86_64_sol2 -r -o
 MKLIB = @MKLIB@
 CONFLIBDIR = @CONFLIBDIR@
 STRIP = @STRIP@
